homeassistant/components/niko_home_control/config_flow.py

```diff
@@ -0,0 +1,72 @@
+"""Config flow for the Niko home control integration."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from nikohomecontrol import NikoHomeControlConnection
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_PORT
+
+from .const import DEFAULT_IP, DEFAULT_PORT, DOMAIN
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST, default=DEFAULT_IP): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): vol.Coerce(int),
+    }
+)
+
+
+def test_connection(host: str, port: int) -> str | None:
+    """Test if we can connect to the Niko Home Control controller."""
+    try:
+        if NikoHomeControlConnection(host, port):
+            return None
+    except Exception:  # noqa: BLE001
```

Comment:
Can we catch more specific exceptions?

Reply:
We can, after the lib is changed, for now it doesn't throw/raise anything

Reply:
Yes we can!